autoapi/mappers/python/parser.py

```diff
@@ -138,14 +141,42 @@ def parse_classdef(self, node, data=None):
                     if not assign_value:
                         continue
                     name = assign_value[0]
-                if not name or name in seen:
+
+                if not name or name in overridden:
                     continue
                 seen.add(name)
                 child_data = self.parse(child)
-                if child_data:
-                    for single_data in child_data:
-                        single_data["inherited"] = base is not node
-                    data["children"].extend(child_data)
+
+                for single_data in child_data:
+                    if single_data["type"] in ("method", "property"):
+                        if name in overloads:
+                            grouped = overloads[name]
+                            if single_data["doc"]:
+                                grouped["doc"] += "\n\n" + single_data["doc"]
```

Comment:
Autodoc picks docstrings only from actual implementations and completely ignores those of overload definitions. This behavior is partly due to technical limitations, but I think it’s somewhat reasonable because we really have no way to access overloads’ docstring at runtime.

The only reason my code concatenates docstrings is to allow us to write documentation in .pyi stubs that cannot have actual implementations. (Autodoc doesn't support .pyi yet.)

In standard .py (not .pyi) code, we shouldn't write docstrings in overload definitions.

Reply:
Let's stick with ignoring the docstrings of overloads and use only the docstring of the actual implementation. It seems like overloads aren't supposed to have docstrings, plus Sphinx doesn't render them that way. Authors then have more control over how the docstring is rendered because they'll be formatting it however they like in the single primary docstring.

Reply:
Autoapi's support for Python stubs (.pyi) is very useful and important for C-extension libraries. (Autodoc is also trying to support it (https://github.com/sphinx-doc/sphinx/pull/4824))

If we completely ignore docstrings of overloads, we will be unable to write docs on Python stubs because we aren't allowed to have actual implementations for overloads in stub files. I'll show an example:

test_invalid.pyi
```python test.pyi
from typing import overload, Union

def hello(a: int) -> int:
    """This is okay"""

@overload
def double(a: int) -> int: ...
@overload
def double(a: str) -> int:
def double(a: Union[str, int]) -> int:
    """DOCSTRING"""
```

run `mypy test.pyi`:

```
test_invalid.pyi:10: error: An implementation for an overloaded function is not allowed in a stub file
Found 1 error in 1 file (checked 1 source file)
```

So we have to write the docstring in an overload.

test_valid.pyi:

```python
from typing import overload, Union

def hello(a: int) -> int:
    """This is okay"""

@overload
def double(a: int) -> int: ...
@overload
def double(a: str) -> int:
    """DOCSTRING"""
```

run `mypy --strict test_valid.pyi`

```
Success: no issues found in 1 source file
```

Do you have any ideas about this?

I think employing only the last docstring is better than concatenating.

Reply:
When using stubgen to generate a stub file for a C extension written with pybind, it sorts the actual implementation to be the last.
So I agree. Let's use only the last docstring when there are no implementations of the function/method.